api/logs/logs.go

```diff
@@ -29,13 +29,14 @@ type SearchBackendConfig struct {
 	File          *FileSearchBackendConfig       `json:"file,omitempty" yaml:"file,omitempty"`
 }
 
+func NewSearchBackend(api SearchAPI) SearchBackend {
+	return SearchBackend{
+		API: api,
+	}
+}
+
 type SearchBackend struct {
-	Name          string
-	API           SearchAPI
-	ElasticSearch *ElasticSearchBackendConfig
-	OpenSearch    *OpenSearchBackendConfig
-	Kubernetes    *KubernetesSearchBackendConfig
-	File          *FileSearchBackendConfig
+	API SearchAPI
```

Comment:
The backend itself doesn't need to store all the configurations since the underlying API already has the config it needs.

Also, didn't see the name being used so removed it.